### PR TITLE
feat: crud posts

### PR DESCRIPTION
**Pull request by stackdumper**

### CODE REVIEW

File: server/src/handlers/posts/create-post.ts [hunk 1]
```diff
@@ -1,19 +1,33 @@
 import { Request, Response } from 'express'
 import { Db } from 'mongodb'
+import { log } from 'util'
```
Comment (stackdumper) [inline server/src/handlers/posts/create-post.ts:3]: Unused import

File: server/src/handlers/posts/create-post.ts [hunk 2]
```diff
@@ -1,19 +1,33 @@
...
+      data: Date.now()
+    })
+    
+    return res.status(200).json({
+      postID: createPost.insertedId
+    })
+
+  } catch (error) {
+    return res.status(404).json({
+      message: 'not implemented',
```
Comment (stackdumper) [inline server/src/handlers/posts/create-post.ts:30]: Misleading status and messagee

File: server/src/handlers/posts/create-post.ts [hunk 3]
```diff
@@ -1,19 +1,33 @@
...
-  // TODO:
-  // 1. create a new post for user
-  // 2. get created post id
-  // 3. return post id
+    const { title, description } = req.body
 
-  return res.status(404).json({
-    message: 'not implemented',
-  })
+    const createPost = await database.collection('posts').insertOne({
```
Comment (stackdumper) [inline server/src/handlers/posts/create-post.ts:18]:
```suggestion
    const result = await database.collection('posts').insertOne({
```

File: server/src/handlers/posts/create-post.ts [hunk 4]
```diff
@@ -1,19 +1,33 @@
...
+      title,
+      description,
+      data: Date.now()
```
Comment (stackdumper) [inline server/src/handlers/posts/create-post.ts:21]:
```suggestion
      date: Date.now()
```

File: server/src/handlers/posts/create-post.ts (hunk 2 above)
Comment (vladenego) [reply server/src/handlers/posts/create-post.ts:30]: Do you mean I haven't logged an error in catch block?

File: server/src/handlers/posts/delete-post.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+/** finds and returns a post by id */
+export const deletePostHandler = (database: Db) => async (
+  req: Request,
+  res: Response,
+) => {
+  const { postId } = req.params
+
+  try {
+    const getPostById = await database
```
Comment (stackdumper) [inline server/src/handlers/posts/delete-post.ts:13]: Nonmeaningful variable name

File: server/src/handlers/posts/delete-post.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      .collection('posts')
+      .findOneAndDelete({ _id: new ObjectId(postId) })
+
+    console.log(getPostById)
+
+    return res.status(200).json({
+      updatedPost: getPostById,
```
Comment (stackdumper) [inline server/src/handlers/posts/delete-post.ts:20]: What is sent to client here?

File: server/src/handlers/posts/delete-post.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    })
+  } catch (error) {
+    return res.status(404).json({
+      message: 'not implemented',
```
Comment (stackdumper) [inline server/src/handlers/posts/delete-post.ts:24]: Misleading status and message

File: server/src/handlers/posts/get-post.ts [hunk 8]
```diff
@@ -1,15 +1,32 @@
...
+  console.log(getPostById);
 
-  return res.status(404).json({
-    message: 'not implemented',
+  return res.status(200).json({
+    post: getPostById
   })
+    
+  } catch (error) {
+    return res.status(404).json({
```
Comment (stackdumper) [inline server/src/handlers/posts/get-post.ts:23]: Misleading status and message

Comment (stackdumper) [reply server/src/handlers/posts/get-post.ts:23]: Also, it's useful to log messages in catch blocks (`console.error`)

File: server/src/handlers/posts/delete-post.ts (hunk 6 above)
Comment (vladenego) [reply server/src/handlers/posts/delete-post.ts:20]: Post in json format

File: server/src/handlers/posts/list-posts.ts [hunk 9]
```diff
@@ -1,13 +1,28 @@
...
   // TODO:
   // 1. find all posts in database
   // 2. return posts
 
-  return res.status(404).json({
-    message: 'not implemented',
-  })
+  try {
+
+    const getAllPosts = await database.collection('posts').find().toArray()
```
Comment (stackdumper) [inline server/src/handlers/posts/list-posts.ts:13]:
```suggestion
    const posts = await database.collection('posts').find().toArray()
```

File: server/src/handlers/posts/update-post.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+export const updatePostHandler = (database: Db) => async (
+  req: Request,
+  res: Response,
+) => {
+  const { postId } = req.params
+
+  try {
+    const getPostById = await database
+      .collection('posts')
+      .findOneAndReplace(
```
Comment (stackdumper) [inline server/src/handlers/posts/update-post.ts:15]:
```suggestion
      .findOneAndUpdate(
```

https://stackoverflow.com/a/39155411/7607265

File: server/src/handlers/posts/update-post.ts [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+/** finds and returns a post by id */
+export const updatePostHandler = (database: Db) => async (
+  req: Request,
+  res: Response,
+) => {
+  const { postId } = req.params
+
+  try {
+    const getPostById = await database
```
Comment (stackdumper) [inline server/src/handlers/posts/update-post.ts:13]:
```suggestion
    const post = await database
```

File: server/src/handlers/posts/create-post.ts (hunk 2 above)
Comment (stackdumper) [reply server/src/handlers/posts/create-post.ts:30]: I mean 404 is wrong code and 'not implemented' is a bad message

File: server/src/handlers/posts/delete-post.ts (hunk 6 above)
Comment (stackdumper) [reply server/src/handlers/posts/delete-post.ts:20]: No need to return deleted pod data, use `.deleteOne` instead

**Comment by stackdumper** [comment]
@vladenego Also, don't forget to format code before committing

**Pull request merged**